draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -0,0 +1,546 @@
+ |                                                          |
+ | +-------------------------+     +--------------------+   |
+ | |                         |     |                    |   |
+ | |   Authorization Server  |     | Protected Resource |   |
+ | |                         |     |                    |   |
+ | +-------^---------+-------+     +----------^---------+   |
+ +---------+---------+------------------------+-------------+
+           |         |                        |
+           |   3) access token           4) access
+           |         |                        |
+2) present assertion |                        |
+           |         |     +------------------+
+           |         |     |
+ +---------+---------+-----+--------------------------------+
+ |         |         |     |             Workload Platform  |
+ |         |         |     |                                |
+ |  +------+---------v-----+---+           +-------------+  |
+ |  |                          |           | Credential  |  |
+ |  |        Workload          +-----------> issued by   |  |
+ |  |                          |  1) get   | Platform    |  |
+ |  +--------------------------+           +-------------+  |
+ +----------------------------------------------------------+
+~~~
+{: #fig-overview title="OAuth2 Assertion Flow in generic Workload Environment"}
+
+The figure outlines the following steps which are applicable in any pattern.
+
+* 1) retrieve credential issued by platform. The way this is achieved and whether this is workload or platform initiated differs based on the platform.
+
+* 2) present credential as an assertion towards an authorization server in an external authorization domain. This step uses the assertion_grant flow defined in {{RFC7521}} and, in case of JWT format, {{RFC7523}}.
+
+* 3) On success, an access token is returned to the workload to access the protected resource.
+
+* 4) The access token is used to access the protected resource in the external authorization domain.
+
+Accessing different protected resources may require steps 2) to 4) again with different scope parameters. Accessing a protected resource in an entirely different authorization domain often requires the entire flow to be followed again, to retrieve a new platform-issued credential with an audience for the external authorization server. This, however, differs based on the platform and implementation.
+
+## Credential format
+
+For the scope of this document we focus on JSON Web Token credential formats. Other formats such as X.509 certificates are possible but not as widely seen as JSON Web Tokens.
+
+The claims in the present assertion vary greatly based on use case and actual platform, but a minimum set of claims are seen across all of them. {{RFC7523}} describes them in detail and according to it, all MUST be present.
+
+~~~json
+{
+  "iss": "https://example.org",
+  "sub": "my-workload",
+  "aud": "custom-audience",
```

Comment:
target-audience?

Reply:
target audience is better than custom audience.